music21/expressions.py

```diff
@@ -517,21 +519,205 @@ def fillListOfRealizedNotes(
+                        alteredPitches=alteredPitches,
+                        cautionaryPitchClass=cautionaryPitchClass,
+                        cautionaryAll=cautionaryAll,
+                        overrideStatus=overrideStatus,
+                        cautionaryNotImmediateRepeat=cautionaryNotImmediateRepeat,
+                        lastNoteWasTied=lastNoteWasTied)
+
+    @property
+    def otherPitches(self) -> tuple[pitch.Pitch, ...]:
+        return self._otherPitches
+
 
 # ------------------------------------------------------------------------------
 class GeneralMordent(Ornament):
     '''
     Base class for all Mordent types.
     '''
-    def __init__(self, **keywords):
+    def __init__(self, *, accid: str | None = None, **keywords):
         super().__init__(**keywords)
-        self.direction = ''  # up or down
-        self.size = None  # interval.Interval (General, etc.) class
+        self._direction = ''  # up or down
+        self._accid: str | None = None
+        if accid is not None and pitch.isValidAccidentalName(accid):
+            self._accid = pitch.makeAccidentalName(accid)
+        self._size: interval.IntervalBase | None = None
         self.quarterLength = 0.125  # 32nd note default
-        self.size = interval.GenericInterval(2)
         self.placement = 'above'
 
-    def realize(self, srcObj: 'music21.note.Note', *, inPlace=False):
+    @property
+    def name(self) -> str:
+        '''
+        returns the name of the Mordent/InvertedMordent, which is generally
+        the class name lowercased, with spaces where a new capital occurs. The
+        name also will include the accidental, if it exists.
+
+        Subclasses can override this as necessary.
+
+        >>> mordent = expressions.Mordent()
+        >>> mordent.name
+        'mordent'
+
+        >>> invertedMordent = expressions.InvertedMordent(accid='sharp')
+        >>> invertedMordent.name
+        'inverted mordent (sharp)'
+
+        '''
+        theName: str = super().name
+        if self.accid is not None:
+            theName += ' (' + self.accid + ')'
+        return theName
+
+    @property
+    def accid(self) -> str | None:
+        return self._accid
+
+    @accid.setter
+    def accid(self, newAccid: str | None):
+        if self._size is not None:
+            raise ExpressionException('Cannot set mordent accid if size already set.')
+
+        # silently ignore invalid accid
+        if newAccid is None:
+            self._accid = None
+        elif pitch.isValidAccidentalName(newAccid):
+            self._accid = pitch.makeAccidentalName(newAccid)
+
+    @property
+    def size(self) -> interval.IntervalBase | None:
+        if self._size is None:
+            raise ExpressionException(
+                'Cannot compute mordent size without srcPitch unless .size has been '
+                + 'explicitly set (use mordent.getSize(srcPitch) instead.)'
+            )
+        return self._size
```

Comment:
property getters are not allowed to raise exceptions as of v7.  The reason?  Many debuggers in IDEs treat properties as the same as attributes and call them all in the process of debugging.  What was wrong with GenericInterval(2) as before?

Reply:
If I return GenericInterval(2), then I have a very common situation (`size` has not been explicitly set) where `size` will incorrectly return GenericInterval(2), and `getSize(src)` will return the correct interval.  That feels _really_ weird to me.  I would almost rather remove the `size` property entirely, and deal with the compatibility consequences (including in ornamentRecognizer).

Reply:
I'm going to return None for now.  We can discuss further.

Reply:
I'm fine with breaking the compatibility on this and removing .size -- since all mordents are generic interval of 2 of some sort anyhow.

Reply:
Oooh.  Done.  I tried to get ornamentRecognizer to work (it was setting/getting trill.size), but I failed.  Take a look at my attempt and see if you can help me.

Reply:
Done (question about ornamentRecognizer is at the top level with more details now).